src/main/java/io/github/nucleuspowered/nucleus/modules/ticket/data/TicketDataCache.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * This file is part of Nucleus, licensed under the MIT License (MIT). See the LICENSE.txt file
+ * at the root of this project for more details.
+ */
+package io.github.nucleuspowered.nucleus.modules.ticket.data;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import io.github.nucleuspowered.nucleus.api.nucleusdata.Ticket;
+import org.apache.commons.lang3.tuple.Triple;
+
+import java.sql.*;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+public class TicketDataCache {
+    private TicketDataManager ticketDataManager;
+    private Map<Integer, TicketData> cachedTickets = Maps.newHashMap(); //TODO Actually cache retrieved data and serve it.
+
+    public TicketDataCache(TicketDataManager ticketDataManager) {
+        this.ticketDataManager = ticketDataManager;
+    }
+
+    public CompletableFuture<List<Ticket>> lookupTicket(List<Triple<String, String, String>> arguments) throws SQLException {
+        CompletableFuture<List<Ticket>> future = new CompletableFuture<>();
+        List<Ticket> result = Lists.newArrayList();
+        //Create a selection query based on the list of Tri's provided. The first should be the column, the second should be the comparator and the third should be the value.
+        String query = "SELECT * FROM Tickets";
+        if (arguments.size() > 0) {
+            query += " WHERE (";
+            Iterator<Triple<String, String, String>> argumentIterator = arguments.iterator();
+            while (argumentIterator.hasNext()) {
+                Triple<String, String, String> argument = argumentIterator.next();
+                query += argument.getLeft() + " " + argument.getMiddle() + " " + argument.getRight();
```

Comment:
This is potentially really dangerous, because I could perform SQL injection very easily via the API - another reason we should be performing all the checking and not exposing implementation details.

Can we do something with prepared statements here?